migration/migrator/migrations/course/20240208120646_custom_answers_polls.py

```diff
@@ -0,0 +1,40 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+        """
+        ALTER TABLE polls
+        ADD COLUMN IF NOT EXISTS allows_custom boolean NOT NULL DEFAULT false;
+
+        ALTER TABLE poll_options
+        ADD COLUMN IF NOT EXISTS author_id VARCHAR(255) NOT NULL DEFAULT '';
```

Comment:
Shouldn't this have a foreign-key constraint too?  We should require new not-null rows to reference a valid student.  If a student is removed, the author_id shouldn't change though.

Reply:
@williamjallen Would adding something like this work? I'm not sure how to test this.

ALTER TABLE poll_options
ADD CONSTRAINT poll_options_fkey
FOREIGN KEY (author_id) REFERENCES public.users(user_id);

Reply:
Done in [335259e](https://github.com/Submitty/Submitty/pull/10197/commits/335259e63f73a2563bf88125f4fba290323c1090)